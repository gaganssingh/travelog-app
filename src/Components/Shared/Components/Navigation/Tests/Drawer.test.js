jest.mock("react-dom");
import React from "react";
import { createPortal } from "react-dom";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import ReactDOM from "react-dom";

class Drawer extends React.Component {
    constructor(props) {
        super(props);
        this.DrawerContainer = document.createElement("div");
    }
    render() {
        return (
            <div>{createPortal(<div>hello</div>, this.DrawerContainer)}</div>
        );
    }
}

test("Drawer renders", () => {
    const component = renderer.create(<Drawer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("Check it returns a div", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Drawer />);
    const result = renderer.getRenderOutput();

    expect(result.type).toBe("div");
    expect(result).toMatchSnapshot();
});
