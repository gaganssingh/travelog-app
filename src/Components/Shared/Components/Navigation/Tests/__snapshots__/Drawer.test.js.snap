// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check it returns a div 1`] = `<div />`;

exports[`Drawer renders 1`] = `<div />`;
